### ISSUE TITLE
Temporary files should be created in $TMPDIR

### ISSUE THREAD

**Issue by zstyblik** [metacloud/molecule#1141]
# Issue Type

- Bug report

# Molecule and Ansible details

`pip install 'molecule==2.8.2'`

# Desired Behaviour

molecule should place temporary files in $TMPDIR and not anywhere in $CWD or $CWD/molecule

# Actual Behaviour (Bug report only)

```
$ docker run -t --rm -v /some/host/path:/source:ro some:image molecule lint
Traceback (most recent call last):
  File "/usr/bin/molecule", line 11, in <module>
    sys.exit(main())
  File "/usr/lib/python2.7/site-packages/click/core.py", line 722, in __call__
    return self.main(*args, **kwargs)
  File "/usr/lib/python2.7/site-packages/click/core.py", line 697, in main
    rv = self.invoke(ctx)
  File "/usr/lib/python2.7/site-packages/click/core.py", line 1066, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/usr/lib/python2.7/site-packages/click/core.py", line 895, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/usr/lib/python2.7/site-packages/click/core.py", line 535, in invoke
    return callback(*args, **kwargs)
  File "/usr/lib/python2.7/site-packages/click/decorators.py", line 17, in new_func
    return f(get_current_context(), *args, **kwargs)
  File "/usr/lib/python2.7/site-packages/molecule/command/lint.py", line 82, in lint
    base.get_configs(args, command_args), scenario_name)
  File "/usr/lib/python2.7/site-packages/molecule/command/base.py", line 121, in get_configs
    _verify_configs(configs)
  File "/usr/lib/python2.7/site-packages/molecule/command/base.py", line 134, in _verify_configs
    scenario_names = [c.scenario.name for c in configs]
  File "/usr/lib/python2.7/site-packages/molecule/config.py", line 201, in scenario
    return scenario.Scenario(self)
  File "/usr/lib/python2.7/site-packages/molecule/scenario.py", line 82, in __init__
    self._setup()
  File "/usr/lib/python2.7/site-packages/molecule/scenario.py", line 170, in _setup
    os.mkdir(self.ephemeral_directory)
OSError: [Errno 30] Read-only file system: '/source/molecule/default/.molecule'
```

Thank you.

**Comment by retr0h** [comment]
Unfortunately this is where molecule stores it's file.  This isn't going to change anytime soon.

**Comment by zstyblik** [comment]
I'm not sure why "wontfix", but ok.

**Comment by retr0h** [comment]
Molecule is built around an ephemeral directory and temp files inside there.  Reworking to $TMPDIR is a larger effort.  Cannot fix this at the moment.

**Comment by retr0h** [comment]
@dj-wasabi interested or have the time to grab this one?

**Comment by elliotweiser** [comment]
@retr0h I can take a crack at this. Seems like a good way to get into the internals.

**Comment by retr0h** [comment]
@elliotweiser  👍

**Comment by retr0h** [comment]
@elliotweiser I ended up having a bit of free time and looked at this.  Please review if you can, and my apologies for stepping on your toes.  Didn't expect I would have time to look at this.

**Comment by elliotweiser** [comment]
@retr0h beat me to it 😄 I can probably give it a test run this weekend.

**Comment by retr0h** [comment]
Now testinfra complains when running Molecule with dind and r/o volume.  However, we no longer have the issue with molecue writing files to $TMPDIR.   I'll look into this separately.

```
py.error.EROFS: [Read-only file system]: mkdir('/tmp/foo/molecule/default/.pytest_cache',)
```

**Comment by retr0h** [comment]
Corrected in 2.12.1 and updated [dind instructions with read-only volume](http://molecule.readthedocs.io/en/latest/examples.html#docker).
